### DISCUSSION TITLE
Grabbing IP Address?

### DISCUSSION THREAD

**Discussion by PhatHub** [microchip-pic-avr-solutions/avr-iot-cellular-arduino-library#14] [Q&A] [answered]
I'm not sure if I'm missing something, but is there a way to get the IP address from LteClass after a successful connection?

**Comment by M65649** [comment, marked as answer]
Hello,

There is an AT command for this, but we haven't implemented it in the LteClass (yet). The AT command is `AT+CGPADDR`. You can test this by uploading the `debug_modem.ino` sketch, open up a serial terminal and type the following (enter after each command):

1. Connect to the network:  `AT+CFUN=1`. Wait until you get +CEREG: 5.
2. Get the IP address: `AT+CGPADDR`.

The documentation about the AT commands can be found in the first link [here](https://iot.microchip.com/docs/arduino/debugging/atcommands). We can implement this functionality in the library for the next release. If you'd like a go at it yourself in the meantime you can see how you can send AT commands to the modem in the `custom_at_commands.ino` example sketch. 

If there is any specific functionality that you need the IP address for, e.g. sockets and such, we're more than happy to hear and we can implement functionality for that directly in the library if you'd like.

**Reply by PhatHub** [reply]
Thanks! I'll check that sketch out!
